packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts

```
@@ -1,147 +1,289 @@
+  organizer,
+  loggedInUserTranslate: translate,
+}: {
+  organizer: {
+    name: string;
+    metadata: {
+      defaultConferencingApp?: NonNullable<UserMetadata>["defaultConferencingApp"];
+    } | null;
+  };
+  /**
+   * translate is used to translate if any error is thrown
+   */
+  loggedInUserTranslate: Awaited<ReturnType<typeof getTranslation>>;
+}) {
+  const organizerMetadata = organizer.metadata;
+  const defaultConferencingApp = organizerMetadata?.defaultConferencingApp;
+  if (!defaultConferencingApp) {
+    throw new UserError(
+      translate("organizer_default_conferencing_app_not_found", { organizer: organizer.name })
+    );
+  }
+  const defaultConferencingAppSlug = defaultConferencingApp.appSlug;
+  const app = getAppFromSlug(defaultConferencingAppSlug);
+  if (!app) {
+    throw new SystemError(`Default conferencing app ${defaultConferencingAppSlug} not found`);
+  }
+  const defaultConferencingAppLocationType = app.appData?.location?.type;
+  if (!defaultConferencingAppLocationType) {
+    throw new SystemError("Default conferencing app has no location type");
   }
+
+  const location = defaultConferencingAppLocationType;
+  const locationType = getEventLocationType(location);
+  if (!locationType) {
+    throw new SystemError(`Location type not found: ${location}`);
+  }
+
+  if (locationType.linkType === "dynamic") {
+    // Dynamic location type need to return the location as it is e.g. integrations:zoom_video
+    return location;
+  }
+
+  const appLink = defaultConferencingApp.appLink;
+  if (!appLink) {
+    throw new SystemError(`Default conferencing app ${defaultConferencingAppSlug} has no app link`);
+  }
+  return appLink;
+}
+
+export async function editLocationHandler({ ctx, input }: EditLocationOptions) {
+  const { newLocation, credentialId: conferenceCredentialId } = input;
+  const { booking, user: loggedInUser } = ctx;
+
+  const organizer = await UserRepository.findByIdOrThrow({ id: booking.userId || 0 });
```

When would the `booking.userId` not be defined? Should we instead throw an error?

---

Yeah, certainly that is the best way.  Added a followup ticket